plugins/Lv2Effect/Lv2Effect.h

```diff
@@ -0,0 +1,54 @@
+﻿/*
+ * Lv2Effect.h - implementation of LV2 effect
+ *
+ * Copyright (c) 2018-2020 Johannes Lorenz <jlsf2013$users.sourceforge.net, $=@>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LV2_EFFECT_H
+#define LV2_EFFECT_H
+
+#include "Effect.h"
+#include "Lv2FxControls.h"
+
+class Lv2Effect : public Effect
+{
+	Q_OBJECT
+
+public:
+	/*
+		initialization
+	*/
+	Lv2Effect(Model* parent, const Descriptor::SubPluginFeatures::Key* _key);
+	//! Must be checked after ctor or reload
+	bool isValid() const { return m_controls.isValid(); }
+
+	bool processAudioBuffer( sampleFrame* buf, const fpp_t frames ) override;
+	EffectControls* controls() override { return &m_controls; }
+
+	Lv2FxControls* lv2Controls() { return &m_controls; }
+	const Lv2FxControls* lv2Controls() const { return &m_controls; }
+
+private:
+	Lv2FxControls m_controls;
+	std::vector<sampleFrame> m_tmpOutputSmps;
```

Comment:
This is invalid because `sampleFrame` is an array type. For some reason, `libstdc++` allows it. This fails to build on macOS(reported on Discord), but the CI was not spotting this because you didn't add LV2 packages to macOS CI builds.

Reply:
Can we redefine `sampleFrame` from `sample_t[DEFAULT_CHANNELS]` to `std::array<sample_t, DEFAULT_CHANNELS>`? The latter is generally recommended over plain arrays in modern C++ as it is much better behaved.

Reply:
@DomClark That's what's done in #4994.

Reply:
Will be fixed now in a new PR #5536 .

@Reflexe you will need to merge this branch (or rebase to master) - If you do, watch out for merge conflicts in lines marked "@recording".